src/main/kotlin/me/partlysanestudios/partlysaneskies/features/economy/auctionhousemenu/AuctionHouseGui.kt

```diff
@@ -216,35 +199,39 @@ class AuctionHouseGui(defaultAuctionInventory: IInventory) : WindowScreen(Elemen
             openMenu()
         }
 
-        fun isAhGui(inventory: IInventory?): Boolean {
-            if (inventory == null) {
-                return true
-            }
-
-            if (PartlySaneSkies.minecraft.currentScreen !is GuiChest) {
-                return false
-            }
-            return inventory.displayName.formattedText
+        fun isAhGui(inventory: IInventory): Boolean = inventory.displayName.formattedText
                 .removeColorCodes()
                 .contains("Auctions Browser") ||
                 inventory.displayName.formattedText
                     .removeColorCodes()
                     .contains("Auctions: \"")
-        }
 
-        private fun openMenu(): AuctionHouseGui {
-            var inventory = (PartlySaneSkies.minecraft.currentScreen as GuiChest).containerInventory
+        private fun openMenu(): AuctionHouseGui? {
+            var attempts = 0
+            val maxAttempts = 5
```

Comment:
Make it a Config option at least?

Reply:
Im not sure if an option is really needed here
this runs anyway every single tick